### PR TITLE
utils: use SystemRandom when generating random password.

### PR DESCRIPTION
**Pull request by xnox**
As noticed by Seth Arnold, non-deterministic SystemRandom should be
used when creating security-sensitive random strings.

LP: #1860795

### CODE REVIEW

**Comment by xnox** [comment]
> just as a comment... there is a long string of issues using random in early boot.
> 
> * https://bugs.launchpad.net/cloud-images/+bug/1584147
> * https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1780062
> * https://bugs.launchpad.net/ubuntu/+source/cloud-init/+bug/1727358
> 
> summary: be careful with random. Especially if you're attempting to get "Better random".

Yes this will cause to block and wait on randomness. However, RANDOM in setting user passwords is not used by default in most cloud types. And recent kernels have more realiable earlier access to devrandom pools, and we try to push our cloud partners to provide hwrng exposed to the virtual machines too.

At the moment, I'd rather boot slowly / fail to boot, than to provision machines with quickly crackable passwords or those that provide a side-channel to establish the machine seed remotely.

**Comment by igalic** [comment]
perhaps we should also add a warning to the `RANDOM` option.

File: cloudinit/util.py [hunk 1]
```diff
@@ -397,9 +397,10 @@ def translate_bool(val, addons=None):
 
 
 def rand_str(strlen=32, select_from=None):
+    r = random.SystemRandom()
```
Comment (OddBloke) [inline cloudinit/util.py]: [The `SystemRandom` docs](https://docs.python.org/3/library/random.html#random.SystemRandom) say that it is "Not available on all systems." Specifically, it is not available on systems where `os.urandom` would raise `NotImplementedError`:

> On a Unix-like system, random bytes are read from the /dev/urandom device. If the /dev/urandom device is not available or not readable, the NotImplementedError exception is raised.

cloud-init is used on BSD systems, so we cannot assume that we are running on top of the Linux kernel. I don't know the state of urandom on BSDs, but I would like to know that we won't be breaking behaviour there.

**Comment by igalic** [comment]
> cloud-init is used on BSD systems, so we cannot assume that we are running on top of the Linux kernel. I don't know the state of urandom on BSDs, but I would like to know that we won't be breaking behaviour there.


most Unix systems these days symlink `/dev/random` to `/dev/urandom`
so if I'm reading this code right, it should just work™, on all the platforms we support
https://github.com/python/cpython/blob/c4cacc8c5eab50db8da3140353596f38a01115ca/Python/bootstrap_hash.c#L434
(this is the code eventually called by os.urandom

**Pull request merged**